httpcore5/src/main/java/org/apache/hc/core5/util/TimeValue.java

```diff
@@ -371,13 +347,43 @@ public int toSecondsIntBound() {
         return asBoundInt(toSeconds());
     }
 
+    public Timeout toTimeout() {
+        return Timeout.of(duration, timeUnit);
+    }
+
     @Override
-    public String toString() {
-        return String.format(Locale.ROOT, "%,d %s", duration, timeUnit);
+    public int compareTo(final TimeValue other) {
+        if (other != null) {
+            final TimeUnit targetTimeUnit = min(other.getTimeUnit());
+            return Long.compare(convert(targetTimeUnit), other.convert(targetTimeUnit));
+        } else {
+            return -Integer.MAX_VALUE;
```

Comment:
This likely violates:

The natural ordering for a class C is said to be consistent with equals if and only if e1.compareTo(e2) == 0 has the same boolean value as e1.equals(e2) for every e1 and e2 of class C. Note that null is not an instance of any class, and e.compareTo(null) should throw a NullPointerException even though e.equals(null) returns false.

Reply:
@michael-o Are you about throwing `NullPointerException` or is this a personal preference?

Reply:
@michael-o I found the reference you were quoting. I removed special handling of `null` parameters.